opal/corelib/error.rb

```diff
@@ -8,13 +9,17 @@ def self.new(message=nil)
       }
 
       err.name = self.$$name;
-      err.$initialize(message);
+      err.$initialize.apply(err, args);
       return err;
     }
   end
+  
+  def self.exception(*args)
+    new(*args)
+  end
 
-  def initialize(message)
-    `self.message = message`
+  def initialize(*args)
+    `self.message = (args.length > 0) ? args[0] : nil`
```

Comment:
This is probably better as `@message, = *args` although I don't know what it would compile to, if it compiles to shit make it ```@message = `...```` so it's a little clearer it's an ivar.